Feat/search/algolia

griest024 opened the pull request:
## PR Checklist

- [ ] Commit message follows our [contributing guidelines](https://github.com/graycoreio/daffodil/blob/develop/CONTRIBUTING.md#commit)
- [ ] Tests added/updated (for bug fixes/features)
- [ ] Documentation added/updated (for bug fixes/features)

## PR Type
<!-- Select the relevant type(s) -->

- [ ] Bug fix
- [ ] Feature
- [ ] Style update
- [ ] Refactor
- [ ] Test
- [ ] Build
- [ ] CI
- [ ] Docs
- [ ] Performance
- [ ] Other (please describe)

## Current behavior
<!-- Describe the current behavior and link to relevant issue -->

Fixes: #3723
Fixes: #3804 

## New behavior
<!-- Describe the changes -->

## Breaking change?

- [ ] Yes
- [ ] No

<!-- If yes, describe impact and migration path -->

## Additional context
<!-- Any other relevant information -->

xelaint commented:
@griest024 

- [x] Retrieve the package that the search result is from and display it under each result
- [x] `No results found` div should only be visible if search field has input and no results are found

griest024 commented:
@xelaint the text color of highlighted items doesn't switch
<img width="860" height="1073" alt="Screenshot_2025-11-07_16-19-29" src="https://github.com/user-attachments/assets/54b4391a-20df-4d11-a01b-210c410a902b" />

griest024 commented:
> @griest024
> 
>     * [ ]  Retrieve the package that the search result is from and display it under each result
> 
>     * [ ]  `No results found` div should only be visible if search field has input and no results are found

the package will have to wait for #4178 but I fixed the other issue

griest024 commented:
@xelaint I've added the package to the API search results. It doesn't look great so it will need some styling love from you

xelaint commented:
<img width="788" height="560" alt="Screen Shot 2025-11-10 at 9 07 15 AM" src="https://github.com/user-attachments/assets/4351c5ef-5122-425f-abfc-a1811a6caf1e" />

The first five results come from the cart package docs, but it's not really clear. Can the package name also be added to these results?

xelaint commented:
<img width="827" height="599" alt="Screen Shot 2025-11-10 at 9 23 55 AM" src="https://github.com/user-attachments/assets/c1354ad6-3a8c-473b-8f6b-5f95c4b636e8" />

Doesn't seem like search history is working. I searched up cart things before this screen. Closed the modal, reopened it, and these are the results that are displayed. When I inspect the code, it seems like these are displayed via the search-results, not search-history.

griest024 commented:
> <img alt="Screen Shot 2025-11-10 at 9 07 15 AM" width="788" height="560" src="https://private-user-images.githubusercontent.com/15809918/512234458-4351c5ef-5122-425f-abfc-a1811a6caf1e.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjI3OTMyMDMsIm5iZiI6MTc2Mjc5MjkwMywicGF0aCI6Ii8xNTgwOTkxOC81MTIyMzQ0NTgtNDM1MWM1ZWYtNTEyMi00MjVmLWFiZmMtYTE4MTFhNmNhZjFlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTEwVDE2NDE0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY1NDBiNTEwOTNhZjkwZGQxNTVhYTgyYmZhN2E4MGU3OTI0ZGEyYmVmMWUwZjg1YTY0ZmI0OWE0ZTI2MGI4ZTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.PKuqcJGUBol0U_ScUZJcVsKH3UtqpWxPXa6-dEkrlyA">
> 
> The first five results come from the cart package docs, but it's not really clear. Can the package name also be added to these results?

hm should `@daffodil/ssr` instead show as `Overview | @daffodil/ssr` then?

xelaint commented:
> > <img alt="Screen Shot 2025-11-10 at 9 07 15 AM" width="788" height="560" src="https://private-user-images.githubusercontent.com/15809918/512234458-4351c5ef-5122-425f-abfc-a1811a6caf1e.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjI3OTMyMDMsIm5iZiI6MTc2Mjc5MjkwMywicGF0aCI6Ii8xNTgwOTkxOC81MTIyMzQ0NTgtNDM1MWM1ZWYtNTEyMi00MjVmLWFiZmMtYTE4MTFhNmNhZjFlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTEwVDE2NDE0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY1NDBiNTEwOTNhZjkwZGQxNTVhYTgyYmZhN2E4MGU3OTI0ZGEyYmVmMWUwZjg1YTY0ZmI0OWE0ZTI2MGI4ZTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.PKuqcJGUBol0U_ScUZJcVsKH3UtqpWxPXa6-dEkrlyA">
> > The first five results come from the cart package docs, but it's not really clear. Can the package name also be added to these results?
> 
> hm should `@daffodil/ssr` instead show as `Overview | @daffodil/ssr` then?

Yeah, I think that makes sense?

griest024 commented:
> <img alt="Screen Shot 2025-11-10 at 9 23 55 AM" width="827" height="599" src="https://private-user-images.githubusercontent.com/15809918/512237981-c1354ad6-3a8c-473b-8f6b-5f95c4b636e8.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjI3OTMyMDMsIm5iZiI6MTc2Mjc5MjkwMywicGF0aCI6Ii8xNTgwOTkxOC81MTIyMzc5ODEtYzEzNTRhZDYtM2E4Yy00NzNiLThmNmItNWY5NWM0YjYzNmU4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTEwVDE2NDE0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY0MzZmZTkwYWQ3OWI1ZjE2NDc1ODc2N2YxMTIzOWJjMjYyMjNjMGZkMjhlMTdmMmYyMTVjOWVmNmY1NzYzNGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.6kLzq_Mfb5jEEj9HormKdGsJKU2HowwqsWjo4TcIgzc">
> 
> Doesn't seem like search history is working. I searched up cart things before this screen. Closed the modal, reopened it, and these are the results that are displayed. When I inspect the code, it seems like these are displayed via the search-results, not search-history.

Its a bit tricky. Searching when typing is an incremental search. Using history from  incremental results in this:
<img width="803" height="1017" alt="Screenshot_2025-11-10_14-33-32" src="https://github.com/user-attachments/assets/0a34def4-087e-4b8b-b87b-6225d7eacef9" />
which doesn't seem like what we want. When should a search be added to the history list?

xelaint commented:
> > <img alt="Screen Shot 2025-11-10 at 9 23 55 AM" width="827" height="599" src="https://private-user-images.githubusercontent.com/15809918/512237981-c1354ad6-3a8c-473b-8f6b-5f95c4b636e8.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjI3OTMyMDMsIm5iZiI6MTc2Mjc5MjkwMywicGF0aCI6Ii8xNTgwOTkxOC81MTIyMzc5ODEtYzEzNTRhZDYtM2E4Yy00NzNiLThmNmItNWY5NWM0YjYzNmU4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTEwVDE2NDE0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY0MzZmZTkwYWQ3OWI1ZjE2NDc1ODc2N2YxMTIzOWJjMjYyMjNjMGZkMjhlMTdmMmYyMTVjOWVmNmY1NzYzNGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.6kLzq_Mfb5jEEj9HormKdGsJKU2HowwqsWjo4TcIgzc">
> > Doesn't seem like search history is working. I searched up cart things before this screen. Closed the modal, reopened it, and these are the results that are displayed. When I inspect the code, it seems like these are displayed via the search-results, not search-history.
> 
> Its a bit tricky. Searching when typing is an incremental search. Using history from incremental results in this: <img alt="Screenshot_2025-11-10_14-33-32" width="803" height="1017" src="https://private-user-images.githubusercontent.com/5853222/512352719-0a34def4-087e-4b8b-b87b-6225d7eacef9.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjI4MDY4MjgsIm5iZiI6MTc2MjgwNjUyOCwicGF0aCI6Ii81ODUzMjIyLzUxMjM1MjcxOS0wYTM0ZGVmNC0wODdlLTRiOGItYjg3Yi02MjI1ZDdlYWNlZjkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTExMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTExMTBUMjAyODQ4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YmM0YmNiYTc3NTRlMTFlNjQyZWVmODQ5YWI5MGNmMjVhNWIzNmFmOTkxNTY4MzRmYjdkNGUwZmIyZGUyNTI3YSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.Xo2XzpOuPj44iIybqRNtEdwT_0u9rrGNYUG8WKq-Tfs"> which doesn't seem like what we want. When should a search be added to the history list?

I think search history should only populate for results that have been navigated to?

xelaint commented:
Accessibility considerations to-do (Elain):
- [ ] ARIA support
- [ ] Check screen reader announcements

griest024 commented:
@xelaint okay package guides now show the package. search history should be fixed as well.

xelaint commented:
@griest024 There's still a list of results (that's not part of search hisory) that show up without any input.

Also, when I search "test" and click on the `provideDaffSearchResultKindFactories` result, I expected `provideDaffSearchResultKindFactories` to be in the recent results, not the word "test".

<img width="781" height="712" alt="Screen Shot 2025-11-12 at 1 48 38 PM" src="https://github.com/user-attachments/assets/ef455e38-0627-4ce7-b192-ba55e88b67bb" />

griest024 commented:
> @griest024 There's still a list of results (that's not part of search hisory) that show up without any input.
> 
> Also, when I search "test" and click on the `provideDaffSearchResultKindFactories` result, I expected `provideDaffSearchResultKindFactories` to be in the recent results, not the word "test".
> 
> <img width="781" height="712" alt="Screen Shot 2025-11-12 at 1 48 38 PM" src="https://github.com/user-attachments/assets/ef455e38-0627-4ce7-b192-ba55e88b67bb" />
> 

Ohh I think I misunderstood. Is there just a "recently clicked on results" list, or is that in addition to the recent *queries* list?

xelaint commented:
> > @griest024 There's still a list of results (that's not part of search hisory) that show up without any input.
> > Also, when I search "test" and click on the `provideDaffSearchResultKindFactories` result, I expected `provideDaffSearchResultKindFactories` to be in the recent results, not the word "test".
> > <img alt="Screen Shot 2025-11-12 at 1 48 38 PM" width="781" height="712" src="https://private-user-images.githubusercontent.com/15809918/513512198-ef455e38-0627-4ce7-b192-ba55e88b67bb.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjI5NzkwOTYsIm5iZiI6MTc2Mjk3ODc5NiwicGF0aCI6Ii8xNTgwOTkxOC81MTM1MTIxOTgtZWY0NTVlMzgtMDYyNy00Y2U3LWIxOTItYmE1NWU4OGI2N2JiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTEyVDIwMTk1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNlZjNhNWQ3NTkxYjM4N2RjN2UxMzkzOTkzNzE4OThlNzFmMjY4NTIxMjRkZjFiODEyYzc0OGVkMTc4MmZjNGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.saZcIA68vyXUGkilt-4i0XWHcC87OUiSzCaOYicUtlU">
> 
> Ohh I think I misunderstood. Is there just a "recently clicked on results" list, or is that in addition to the recent _queries_ list?

There isn't a separate UI for recent queries vs recently clicked on. I think we just want recently clicked, not recent queries.

The screenshot above seems like it's showing the recent queries using the `<div class="daffio-docs-search-history">` container, and then the bottom portion (starting from "Testing") is using this block:

```
@let results = docsResults$ | async;
@let loading = loading$ | async;
@if (results?.length > 0 || loading) {
	<ul class="daffio-docs-search-results" role="listbox"></ul>
```

This block should only ever be visible if the search field has a value and results related to that value.

<img width="611" height="515" alt="Screen Shot 2025-11-12 at 2 20 28 PM" src="https://github.com/user-attachments/assets/4960aaa6-14f9-4bb9-a418-6e82d29a5ae4" />

griest024 commented:
@xelaint okay I've made updates that I think are what you want. We should probably move the result item to its own component

xelaint commented:
@griest024 I tried to move the search results and search history to separate components but couldn't get the `ActiveDescendantKeyManager` to work properly.

Can we also pass in the enter key to store result?

griest024 committed:
feat(core): add injectable action support

griest024 committed:
feat(core)!: modify config token factory to only require props not in default

griest024 committed:
feat(docs): rename doc item factory

griest024 committed:
feat(core): add `group` util

griest024 committed:
feat(docs): add redux state

griest024 committed:
build(search): fix schema path

griest024 committed:
feat(search): extend search options to be a collection request

currently this is only supported by the kinds (product, category) but it makes sense to have for the base driver call as well

griest024 committed:
feat(docs): add algolia search driver

griest024 committed:
chore: add `@algolia/client-search` dep

griest024 committed:
feat(search-docs): add `@daffodil/search-docs` glue package

griest024 committed:
feat(search-docs): add `@daffodil/search-docs/state`

griest024 committed:
feat(search-docs): add `@daffodil/search-docs/driver/algolia`

xelaint commented:
@griest024 What is this waiting on?

griest024 commented:
> @griest024 What is this waiting on?

its waiting on @damienwebdev to give feedback on injectable actions

damienwebdev commented:
@griest024 can you extract the injectable actions into a separate PR?

griest024 committed:
feat(daffio): add search

Co-authored-by: xelaint

griest024 commented:
@xelaint the header is messed up, its possible I did not resolve conflicts correctly, can you take a look?

xelaint committed:
fix header

xelaint commented:
@griest024 fixed
